### PR TITLE
[8.0] add account_reset_chart

### PR DESCRIPTION
**Pull request by StefanRijnhart**
# Account chart reset

Adds a method to the company to remove its chart of accounts, including all
related transactions, journals etc. By necessity, this process also removes
the company's bank accounts as they are linked to the company's journals and
the company's payment orders and payment modes if the payment module is
installed.

As a result, you can then reconfigure the company chart of account with the
same or a different chart template.
# Usage

To prevent major disasters when this module is installed, no interface is
provided. Please run through xmlrpc, for instance using erppeek: ::

```
import erppeek

host = 'localhost'
port = '8069'
admin_pw = 'admin'
dbname = 'openerp'

client = erppeek.Client('http://%s:%s' % (host, port))
client.login('admin', admin_pw, dbname)
client.execute('res.company', 'reset_chart', 1)
```
# Known issues / Roadmap

This should work with the standard accounting modules installed. All sorts of
combinations with third party modules are imaginable that would require
modifications or extensions of the current implementation.

Sequences are not reset during the process.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
We all have this case one time at least :wink: !

Some comments:
- Don't you think this is better in server-tools repo, as it has no interface?
- Reduce icon resolution to 128x128 or so to be manageable (very good icon BTW).

**Comment by StefanRijnhart** [comment]
Good, I hoped you'd like it ;-). I shrunk the icon to 128x128 now.

About the repo: I thought that it should count as a financial tool, not a server wide functionality.

**Comment by pedrobaeza** [comment]
I suppose that try/except blocks are because of some troubles you have found on installations, so it's not a special problem.

:+1:

**Comment by StefanRijnhart** [comment]
Not really troubles, but KeyErrors are caught because they signal the absence of optional modules, e.g. account_payment, account_voucher.

File: account_reset_chart/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                [('company_id', '=', self.id),
+                 ('state', 'in', ('proforma', 'posted'))])
+            vouchers.cancel_voucher()
+            voucher_obj.search(
+                [('company_id', '=', self.id)]).unlink()
+        except KeyError:
+            pass
+
+        try:
+            self.env['payment.order']
```
Comment (eLBati) [inline account_reset_chart/models/res_company.py]: As per travis:

```
account_reset_chart/models/res_company.py:69: [W0104(pointless-statement), Company.reset_chart] Statement seems to have no effect
```

**Comment by StefanRijnhart** [comment]
@eLBati Thanks. Travis is happy again.

**Comment by eLBati** [comment]
LGTM :+1:

File: account_reset_chart/models/res_company.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            self._cr.execute(
+                """UPDATE account_move SET state = 'draft'
+                   WHERE id IN %s""", (tuple(moves.ids),))
+        moves.unlink()
+
+        unlink_from_company('account.fiscal.position')
+        unlink_from_company('account.tax')
+        unlink_from_company('account.tax.code')
+        unlink_from_company('account.journal')
+
```
Comment (JayVora-SerpentCS) [inline account_reset_chart/models/res_company.py:143]: Just an idea..redefine the unlink_from_company as:
def unlink_from_company(models):
....Call the same code but with loop for model in models....
The Call then should be : unlink_from_company([])...less lines and nice too!

**Pull request merged**